### ISSUE TITLE
Lacks TypeScript definitions

### ISSUE THREAD

**Issue by vsviridov** [zerodevx/svelte-toast#1]
With TypeScript being a [first-class Svelte citizen](https://svelte.dev/blog/svelte-and-typescript) it would be great if the TypeScript definitions were provided with the library.

**Comment by vsviridov** [comment]
something along the lines of
```ts
// index.d.ts
declare module "@zerodevx/svelte-toast";

import type { SvelteComponent } from "svelte";
import type { Readable } from "svelte/store";

export interface SvelteToastOptions {
  duration: number = 4000;
  dismissable: boolean = true;
  initial: number;
  progress: number;
  reversed: boolean = false;
  intro: any;
  theme: { [key: string]: string }
}

export class SvelteToast extends SvelteComponent {
  options: SvelteToastOptions;
}

declare namespace toast {
  export function push(text: string, options?: SvelteToastOptions): number;
  export function pop(id: number): void;
  export function set(id: number, options: SvelteToastOptions): void;
  export function _opts(options: SvelteToastOptions = {}): SvelteToastOptions;
  export const subscribe: typeof Readable["subscribe"]
}
```

**Comment by zerodevx** [comment]
Thank you for raising this. Would you be interested in making a PR for this? 🙏

**Comment by zerodevx** [comment]
Will be happy to provide any info that you need.

**Comment by zerodevx** [comment]
Closed with #14
